tree.cpp

```diff
@@ -1,7 +1,77 @@
 #include "tree.h"
-
+#include <limits>
+#include <algorithm>
+#include <iostream>
 
 tree::tree(vector<node> nodes, vector<edge> edges) {
     this->nodes = nodes;
     this->edges = edges;
-}
\ No newline at end of file
+}
+
+float tree::forward_backward_min_marginals(){
+    int nb_nodes = nodes.size();
+
+    // Do the forward
```

Comment:
so a function "forward until" and a "backward until", taking a node id as argument?